contracts/coordinator/src/state.rs

```diff
@@ -12,17 +12,72 @@ pub struct Config {
 
 pub const CONFIG: Item<Config> = Item::new("config");
 
-type ProverAddress = Addr;
+pub type ProverAddress = Addr;
 
 pub const PROVER_PER_CHAIN: Map<ChainName, ProverAddress> = Map::new("prover_per_chain");
 
-// TODO: migrate this?
-pub const ACTIVE_VERIFIER_SET_FOR_PROVER: Map<ProverAddress, VerifierSet> =
-    Map::new("active_prover_verifier_set");
-
-type ChainNames = HashSet<ChainName>;
-type VerifierAddress = Addr;
+pub type ChainNames = HashSet<ChainName>;
+pub type VerifierAddress = Addr;
 pub const CHAINS_OF_VERIFIER: Map<VerifierAddress, ChainNames> = Map::new("chains_of_verifier");
 
-pub const NEXT_VERIFIER_SET_FOR_PROVER: Map<ProverAddress, VerifierSet> =
-    Map::new("next_prover_verifier");
+pub struct VerifierSetIndex<'a> {
+    pub by_verifier: MultiIndex<'a, Addr, VerifierProverRecord, (Addr, Addr)>,
```

Comment:
where is this index actually used? Every place I saw the code is just reading/writing to the map directly, or iterating based on prefix. Is the index used?